src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,195 @@
+
+    let searchTerms: string = `name=${animalLookUp}`;
+    const animalIndex = this._animals.findIndex((animal) => animal.species === animalLookUp);
+    const species = this._findAnimalsFromSubspecies(animalLookUp)?.species;
+
+    // Checks if it's a sub-species, if so it retrieves its species.
+    if (species !== undefined) {
+      searchTerms = `name=${species}&sub-species=${animalLookUp}`;
+    } else if (this._animals[animalIndex].subSpecies.length > 0) {
+      // Detects if the species has subspecies.  Then applies a random sub-species accordingly.
+      const subSpeciesList: string[] = this._animals[animalIndex].subSpecies;
+      const randomSubspecies = subSpeciesList[Math.floor(Math.random() * subSpeciesList.length)];
+
+      searchTerms = `name=${animalLookUp}&sub-species=${randomSubspecies}`;
+    }
+
+    this.container.httpService
+      .get(`${this._API_URL}species/?${searchTerms}`)
+      .then(async (response: IHttpResponse) => {
+        // Notifies the user if there was a problem contacting the server
+        if (Math.floor(response.status / 100) !== 2) {
+          message.reply(
+            `Something seems to have happened with the connection to ${this._API_URL}.`
+          );
+          return;
+        }
+
+        const [animalData] = response.data.animal;
+        await message.reply('', { files: [animalData.Image], name: 'image.jpg' });
+      })
+      .catch((err) => {
+        this.container.loggerService.warn(err);
+      });
+  }
+
+  private async _pickListType(message: IMessage, input?: string) {
+    if (input === undefined || input.startsWith('species')) {
+      message.reply(this._makeSpeciesEmbed());
+      return;
+    }
+
+    if (input && input.startsWith('subspecies')) {
+      message.reply(this._makeSubspeciesEmbed());
+      return;
+    }
+
+    await message.reply(this._makeSingleSubSpeciesEmbed(input));
+    return;
+  }
+
+  private _getRandomAnimal(): string {
+    const animals: string[] = Array.from(this._validAnimals);
+    return animals[Math.floor(Math.random() * animals.length)];
+  }
+
+  private _makeSpeciesEmbed() {
+    if (this._animalEmbed || this._animals.length < 1) {
+      return this._animalEmbed ?? 'Animal API has not loaded yet.';
+    }
+
+    const species: string[] = this._animals.map((animal: ISubSpecies) => animal.species);
+    this._animalEmbed = this.container.messageService.generateEmbedList(species);
+    this._animalEmbed.setColor('#0099ff').setTitle('Species');
+
+    return this._animalEmbed;
+  }
+
+  private _makeSubspeciesEmbed(): MessageEmbed {
+    if (this._subspeciesEmbed) {
+      return this._subspeciesEmbed;
+    }
+
+    const embed = new MessageEmbed();
+    embed.setColor('#0099ff').setTitle('Subspecies');
+
+    this._animals
+      .filter((animal) => !!animal.subSpecies)
+      .forEach((animal) => {
+        embed.addField(animal.species, animal.subSpecies.join('\n'), true);
+      });
+
+    return (this._subspeciesEmbed = embed);
```

Comment:
its not expensive computationally to create this embed, it would simplify it a lot if you did not cache it

Reply:
and same for the rest of the embeds